### PR TITLE
Edit and Remove item from Cart

### PR DESCRIPTION
**Pull request by trinhdvt**
## Related Tickets
- [#46911](https://edu-redmine.sun-asterisk.vn/issues/46911)

## WHAT (optional)
- Refactor code
- Implement change product' quantity in cart
- Implement delete product from cart

## HOW
- At Cart detail page, use can click on `-` / `+` button or type in new quantity for product. An `ajax PUT` request will be sent to update cart in cookie and also update the UI.
- When user click on `delete` button, an `ajax DELETE` request will be sent to remove specific product from cart or delete them all.

## Evidence (Screenshot or Video)
Ảnh chụp Màn hình 2022-03-17 lúc 20 09 52

Ảnh chụp Màn hình 2022-03-17 lúc 20 36 13

Ảnh chụp Màn hình 2022-03-17 lúc 20 35 30

### CODE REVIEW

File: app/views/carts/_cart.html.erb [hunk 1]
```diff
@@ -30,26 +33,33 @@
...
           <span class="glyphicon glyphicon-minus"></span>
         </button>
 
-        <input type="text" class="quantity_input text-center full-width" value="<%= @quantity[idx] %>">
+        <input type="text" class="quantity_input text-center full-width"
+               value="<%= @quantity[idx] %>" id="input-qtt-<%= prod.id %>"
+        >
 
-        <button id="btn-qtt_incr" class="quantity_btn" type="button">
+        <button pid="<%= prod.id %>" class="quantity_btn btn-qtt_incr" type="button">
```
Comment (lcko1012) [inline app/views/carts/_cart.html.erb:44]:
```suggestion
        <button id="<%= prod.id %>" class="quantity_btn btn-qtt_incr" type="button">
```

**Review by lcko1012 — COMMENTED** [review]
cmt

File: app/views/carts/_cart.html.erb (hunk 1 above)
Comment (trinhdvt) [reply app/views/carts/_cart.html.erb:44]: sử dụng `pid` để lưu `product_id` thay vì sử dụng `id` để tránh xung đột với các element khác

**Review by lcko1012 — APPROVED** [review]

**Comment by lcko1012** [comment]
ok

**Comment by trinhdvt** [comment]
ready

**Pull request merged**